cloud/src/meta-service/meta_service.cpp

```diff
@@ -731,6 +731,27 @@ void internal_create_tablet(const CreateTabletsRequest* request, MetaServiceCode
     std::string key;
     std::string val;
     meta_tablet_key(key_info, &key);
+
+    if (txn->get(key, &val) == TxnErrorCode::TXN_OK) {
+        doris::TabletMetaCloudPB exists_tablet_meta;
+        if (!exists_tablet_meta.ParseFromString(val)) {
+            code = MetaServiceCode::PROTOBUF_SERIALIZE_ERR;
+            msg = fmt::format("failed to serialize tablet meta, key={}", hex(key));
+            return;
+        }
+        if (exists_tablet_meta.tablet_id() == tablet_meta.tablet_id() &&
+            exists_tablet_meta.schema_version() == tablet_meta.schema_version()) {
+            // idempotent
+            code = MetaServiceCode::OK;
+            msg = "tablet already exists";
+            LOG(INFO) << "tablet already exists, "
+                      << " request tablet's tablet_id=" << tablet_meta.tablet_id()
+                      << " request tablet's schema_version=" << tablet_meta.schema_version()
+                      << " existed tablet's tablet_id=" << exists_tablet_meta.tablet_id()
+                      << " existed tablet's schema_version=" << exists_tablet_meta.schema_version();
+            return;
+        }
```

Comment:
LOG warning or error, it is unexpected the existing value does not match with the request.